qiskit_research/utils/convenience.py

```diff
@@ -38,24 +38,53 @@ def add_dynamical_decoupling(
     dd_str: str,
     scheduler: BaseScheduler = ALAPScheduleAnalysis,
     add_pulse_cals: bool = False,
+    urdd_pulse_num: int = 4,
+    urdd_pulse_method: Optional[str] = "phase_shift",
 ) -> Union[QuantumCircuit, List[QuantumCircuit], List[List[QuantumCircuit]]]:
     """Add dynamical decoupling sequences and calibrations to circuits.
 
     Adds dynamical decoupling sequences and the calibrations necessary
     to run them on an IBM backend.
+
+    Args:
+        circuits (Union[QuantumCircuit, List[QuantumCircuit], List[List[QuantumCircuit]]]):
+            input QuantumCircuit or sequences thereof.
+        backend (Backend): Backend to run on; gate timing is required for this method.
+        dd_str (str): String describing DD sequence to use.
+        scheduler (BaseScheduler, optional): Scheduler, defaults to ALAPScheduleAnalysis.
+        add_pulse_cals (bool, optional): Add Pulse calibrations for non-basis
+            gates? Defaults to False.
+        urdd_pulse_num (int, optional): URDD pulse number must be even and at least 4.
+            Defaults to 4.
+        urdd_pulse_method (Optional[str], optional): Pulse calibration method for URDD
+            sequences (if add_pulse cals). Defaults to "phase_shift".
```

Comment:
Document the options for `urdd_pulse_method`

Reply:
Created an `enum` class `PulseMethod` to define options, and expanded the docstring.